internal/tsetse/language_service_plugin.ts

```diff
@@ -13,22 +12,28 @@ function init() {
   return {
     create(info: ts.server.PluginCreateInfo) {
       const oldService = info.languageService;
-      const checker = new Checker(oldService.getProgram());
+      const proxy = pluginApi.createProxy(oldService);
+      const program = oldService.getProgram();
 
-      // Add disabledRules to tsconfig to disable specific rules
-      // "plugins": [
-      //   {"name": "...", "disabledRules": ["equals-nan"]}
-      // ]
-      registerRules(checker, info.config.disabledRules || []);
+      // Signature of `getProgram` is `getProgram(): Program | undefined;` in ts 3.1
+      // so we must check if the return value is valid to compile with ts 3.1.
+      if (program) {
```

Comment:
I suspect an undefined `program` would cascade and things would fail anyway so maybe the best thing to do is to throw in this case?

Reply:
Yeah throwing seems the correct behaviour here.

Reply:
👍